python-checks/src/main/java/org/sonar/python/checks/SecureModeEncryptionAlgorithmsCheck.java

```diff
@@ -0,0 +1,135 @@
+    CRYPTODOMEX_VULNERABLE_MODES_BY_API.put("Cryptodome.Cipher.DES3.new", Arrays.asList("Cryptodome.Cipher.DES3.MODE_ECB", "Cryptodome.Cipher.DES3.MODE_CBC"));
+    CRYPTODOMEX_VULNERABLE_MODES_BY_API.put("Cryptodome.Cipher.ARC2.new", Arrays.asList("Cryptodome.Cipher.ARC2.MODE_ECB", "Cryptodome.Cipher.ARC2.MODE_CBC"));
+    CRYPTODOMEX_VULNERABLE_MODES_BY_API.put("Cryptodome.Cipher.Blowfish.new", Arrays.asList("Cryptodome.Cipher.Blowfish.MODE_ECB", "Cryptodome.Cipher.Blowfish.MODE_CBC"));
+    CRYPTODOMEX_VULNERABLE_MODES_BY_API.put("Cryptodome.Cipher.AES.new", Arrays.asList("Cryptodome.Cipher.AES.MODE_ECB", "Cryptodome.Cipher.AES.MODE_CBC"));
+  }
+
+  @Override
+  public void initialize(Context context) {
+    context.registerSyntaxNodeConsumer(CALL_EXPR, ctx -> {
+      CallExpression callExpression = (CallExpression) ctx.syntaxNode();
+      Symbol calleeSymbol = callExpression.calleeSymbol();
+      if (calleeSymbol == null) {
+        return;
+      }
+      checkPycaLibrary(ctx, callExpression, calleeSymbol);
+      checkPycryptoAndCryptodomeLibraries(ctx, callExpression, calleeSymbol);
+      checkPydesLibrary(ctx, callExpression, calleeSymbol);
+    });
+  }
+
+  private static void checkPydesLibrary(SubscriptionContext ctx, CallExpression callExpression, Symbol calleeSymbol) {
+    if ("pyDes.des".equals(calleeSymbol.fullyQualifiedName())) {
+      ctx.addIssue(callExpression.callee(), MESSAGE);
+    }
+  }
+
+  private static void checkPycryptoAndCryptodomeLibraries(SubscriptionContext ctx, CallExpression callExpression, Symbol calleeSymbol) {
+    if ("Crypto.Cipher.PKCS1_v1_5.new".equals(calleeSymbol.fullyQualifiedName()) || "Cryptodome.Cipher.PKCS1_v1_5.new".equals(calleeSymbol.fullyQualifiedName())) {
+      ctx.addIssue(callExpression.callee(), MESSAGE);
+      return;
+    }
+
+    List<String> vulnerableModes = PYCRYPTO_VULNERABLE_MODES_BY_API.getOrDefault(calleeSymbol.fullyQualifiedName(),
+      CRYPTODOMEX_VULNERABLE_MODES_BY_API.get(calleeSymbol.fullyQualifiedName()));
+
+    if (vulnerableModes != null) {
+      List<Argument> arguments = callExpression.arguments();
+      if (arguments.size() < 2) {
+        return;
+      }
+      Argument argument = arguments.get(1);
+      if (argument.is(REGULAR_ARGUMENT)) {
+        Optional<Symbol> symbol = TreeUtils.getSymbolFromTree(((RegularArgument) argument).expression());
+        symbol.filter(s -> vulnerableModes.contains(s.fullyQualifiedName())).ifPresent(s -> ctx.addIssue(argument, MESSAGE));
+      }
+    }
+  }
+
+  protected void checkPycaLibrary(SubscriptionContext ctx, CallExpression callExpression, Symbol calleeSymbol) {
+    if (PYCA_ALGORITHM_METHODS.contains(calleeSymbol.fullyQualifiedName())) {
+      List<Argument> arguments = callExpression.arguments();
+      if (arguments.size() < 2) {
+        return;
+      }
+      Argument argument = arguments.get(1);
```

Comment:
Same here, it could be a keyword argument. 
Not sure we need to cover it though, but if you eventually decide to cover it, don't forget to separate `mode` and `padding` for Pyca.